packages/bierzo-wallet/src/routes/payment/index.tsx

```diff
@@ -54,13 +54,18 @@ const Payment = (): JSX.Element => {
 
     const chainId = token.chainId;
 
-    let recipient: Address;
+    let recipient: LookupResultType;
     if (isIov(formValues[ADDRESS_FIELD])) {
-      try {
-        recipient = await lookupRecipientAddressByName(formValues[ADDRESS_FIELD], chainId);
-      } catch (err) {
-        toast.show(err.message, ToastVariant.ERROR);
-        console.error(err);
+      recipient = await lookupRecipientAddressByName(formValues[ADDRESS_FIELD], chainId);
+
+      if (recipient === "name_not_found") {
+        toast.show("Recipient's personalized address was not found", ToastVariant.ERROR);
+        return;
+      } else if (recipient === "no_address_for_blockchain") {
+        toast.show(
+          "Recipient's personalized address does not contain an address for this blockchain",
+          ToastVariant.ERROR,
+        );
         return;
       }
```

Comment:
and assign `recipient = lookupResult;` only in the case where you know it is an address